ignite/metrics/epoch_metric.py

```diff
@@ -46,8 +46,10 @@ def __init__(self, compute_fn: Callable, output_transform: Callable = lambda x:
         self.compute_fn = compute_fn
 
     def reset(self) -> None:
-        self._predictions = torch.tensor([], dtype=torch.float32)
-        self._targets = torch.tensor([], dtype=torch.long)
+        self._predictions = []
+        self._prediction_tensor = None
```

Comment:
I think we can avoid double storage of predictions and targets by removing `self._prediction_tensor ` and `self._target_tensor` and adapting impacted tests. What do you think ?